### ISSUE TITLE
7.13 "_interopRequireDefault is not a function"

### ISSUE THREAD

**Issue by KyleAMathews** [babel/babel#12854]
In addition to https://github.com/babel/babel/issues/12851

If you set `resolutions` for `@babel/core` to `~7.12.0` you then get this error:

```
   8 | exports.navigateTo = exports.replace = exports.push = exports.navigate = exports.default = void 0;
   9 |
> 10 | var _objectWithoutPropertiesLoose2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutPropertiesLoose"));
     | ^
  11 |
  12 | var _assertThisInitialized2 = _interopRequireDefault(require("@babel/runtime/helpers/assertThisInitialized"));
  13 |


  WebpackError: TypeError: _interopRequireDefault is not a function

  - index.js:10
    node_modules/gatsby-link/index.js:10:1

  - utils.js:136
    node_modules/@reach/router/lib/utils.js:136:1
```

`gatsby-link` requires `babel/runtime/helpers/interopRequireDefault` which seems to no longer exist in `@babel/runtime@7.13`

**Comment by nicolo-ribaudo** [comment]
Niiice we have a ton of tests in theory for this.

**Comment by KyleAMathews** [comment]
well that's strange... https://unpkg.com/@babel/runtime@7.13.0/helpers/interopRequireDefault/index.js

Lemme look again

**Comment by hzoo** [comment]
Think we tried to change to a folder in a backwards compatible way

https://unpkg.com/browse/@babel/runtime@7.12.18/helpers/interopRequireDefault.js vs
https://unpkg.com/browse/@babel/runtime@7.13.0/helpers/interopRequireDefault/index.js

**Comment by nicolo-ribaudo** [comment]
This is the same as https://github.com/babel/babel/issues/12853

**Comment by nicolo-ribaudo** [comment]
@KyleAMathews I know it's probably easy to reproduce this, but do you have a project which already shows the error I can start from? :sweat_smile:

**Comment by KyleAMathews** [comment]
@nicolo-ribaudo use the gatsby default again but with the fix in https://github.com/babel/babel/pull/12852 applied but run `gatsby build` this time :-D

**Comment by nicolo-ribaudo** [comment]
Ok thanks :joy:

**Comment by gillkyle** [comment]
@nicolo-ribaudo you can use this repo! https://github.com/gillkyle/babel-repro

It's got even less packages than the default starter if you're looking for something you can just clone.

I'm getting this errors locally when running gatsby build:

```
failed Building production JavaScript and CSS bundles - 0.973s

 ERROR #98123  WEBPACK

Generating JavaScript bundles failed

Cannot read property 'resolved' of undefined

File: .cache/polyfill-entry.js


 ERROR #98123  WEBPACK

Generating JavaScript bundles failed

Cannot read property 'resolved' of undefined

File: .cache/production-app.js
```

**Comment by KyleAMathews** [comment]
>realizing now how useless that output is (besides file paths)

yeah, we kill internal stack traces which isn't so helpful when that's where they're happening.

**Comment by nicolo-ribaudo** [comment]
I know the problem and the fix, working on the PR.

Babel injects this code:

```js
var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");
```

Webpack sees [this exports map](https://github.com/babel/babel/blob/e3f090395ab34aec3a826e7de571f50aac368c52/packages/babel-runtime/package.json#L150)

And replaces it with 
```js
var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault/index.mjs");
```

**Comment by JLHwung** [comment]
This issue is not reproducible on Webpack 5, which supports exports map. For Webpack 4, a workaround is to configure [`resolve.extensions`](https://webpack.js.org/configuration/resolve/#resolveextensions) to
```js
['.wasm', '.js', '.mjs', '.json']
```
So webpack will resolve `@babel/runtime/helpers/interopRequireDefault` to `@babel/runtime/helpers/interopRequireDefault/index.js`.

**Comment by dlong500** [comment]
This bug is NOT fixed (as evidenced by the many new issues being opened within the last few hours).  It would appear that a LOT of dependencies are pulling in the latest versions of babel automatically which makes it difficult to even temporarily roll back to an earlier babel version (unless you commit a yarn.lock file).  Don't mean to complain--just trying to get the word out that there are still some serious issues unresolved.

**Comment by fantasticsoul** [comment]
I HAVE MET THIS PROBLEM TOO, SOMEBODY HELP ME PLEASE

**Comment by nicolo-ribaudo** [comment]
Could you two share your lockfiles please? 🙏

**Comment by AndriusBil** [comment]
@nicolo-ribaudo 
https://github.com/AndriusBil/babel-require-example

**Comment by JoeeeeeWu** [comment]
_interopRequireDefault is not a function

this bug is not solved

@nicolo-ribaudo could you fixed it as soon as possible?

**Comment by nicolo-ribaudo** [comment]
Yes, I'll start working on this in 10 mins

**Comment by wangcherry** [comment]
> In addition to #12851
> 
> If you set `resolutions` for `@babel/core` to `~7.12.0` you then get this error:
> 
> ```
>    8 | exports.navigateTo = exports.replace = exports.push = exports.navigate = exports.default = void 0;
>    9 |
> > 10 | var _objectWithoutPropertiesLoose2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutPropertiesLoose"));
>      | ^
>   11 |
>   12 | var _assertThisInitialized2 = _interopRequireDefault(require("@babel/runtime/helpers/assertThisInitialized"));
>   13 |
> 
> 
>   WebpackError: TypeError: _interopRequireDefault is not a function
> 
>   - index.js:10
>     node_modules/gatsby-link/index.js:10:1
> 
>   - utils.js:136
>     node_modules/@reach/router/lib/utils.js:136:1
> ```
> 
> `gatsby-link` requires `babel/runtime/helpers/interopRequireDefault` which seems to no longer exist in `@babel/runtime@7.13`

Are you ok? I am error now

**Comment by nicolo-ribaudo** [comment]
@wangcherry Could you upload your lockfile? :pray:

**Comment by dlong500** [comment]
I haven't tested any new developments yet since early this morning (EST).
Here is my yarn.lock in case it is still useful (note that this lock file is with my immediate babel dependences locked to 7.12.x, so only the nested dependencies are using later babel versions).
[yarn.lock.txt](https://github.com/babel/babel/files/6030609/yarn.lock.txt)

**Comment by dlong500** [comment]
Ok, it looks like at least my particular issue is resolved with the latest babel release.  After doing a clean install (wiping node_modules and the lock file) everything seems to be working again.

**Comment by gglee89** [comment]
Hopefully this isn't gibberish from my end.
But I'm getting the "same or similar" error in the browser at this point:
```
Uncaught TypeError: _interopRequireWildcard is not a function at Object../node_modules/@material-ui/icons/ExpandMore.js
```

I've tried running a clean build my React app with these settings:
```json
"dependencies": {
    "@material-ui/core": "^4.11.3",
    "@material-ui/icons": "^4.11.2",
    "react": "^17.0.1",
    "react-dom": "^17.0.1",
},
"devDependencies": {
    "@babel/core": "^7.13.1",
    "@babel/plugin-transform-runtime": "^7.13.6",
    "@babel/preset-env": "^7.13.5",
    "@babel/preset-react": "^7.12.13",
    "webpack": "^5.4.0",
    "webpack-bundle-analyzer": "^4.1.0",
    "webpack-cli": "^4.2.0",
    "webpack-dev-server": "^3.11.0",
    "webpack-merge": "^5.4.0",
}
```

Also tried running @babel with version @7.12, but I got the same error in the browser.

**Comment by nicolo-ribaudo** [comment]
Can you share your lockfile?

**Comment by gglee89** [comment]
Sorry to bother! 

I got it fixed, but I had to install `npm install --save-dev @babel/runtime` to solved it.
The strange is that I didn't need install it before until today. I don't know what's changed.

@nicolo-ribaudo Let me know if you still need to see my Lockfile.
@AndriusBil your [comment](https://github.com/babel/babel/issues/12854#issuecomment-784013285) was what made me attempt to install it. Thanks!

**Comment by nicolo-ribaudo** [comment]
Oh I think that the reason was that one of your dependency was using `@babel/runtime@7.13.0` as a dependency, and by explicitly installing it you forced npm to update it. You can probably remove `@babel/runtime` from your deps now.

**Comment by matyasf** [comment]
Hi!
Is this issue fixed in 7.13? If yes which version? Thanks!

**Comment by nicolo-ribaudo** [comment]
7.13.10 should be working.
